src/ca/mcgill/ecse211/lab4/Lab4.java

```diff
@@ -0,0 +1,109 @@
+
+/**
+ * This class contains the main method which implements the hardware required to execute the fourth
+ * laboratory. The purpose of laboratory 4 is to localize the robot given an arbitrary starting
+ * position. In other words, it is placed at a starting position, and it has to gather data using
+ * the ultrasonic sensor and the light sensor to orient itself towards the origin of the grid
+ * system, i.e (0,0). A reference system is defined in terms of x,y and theta in order to localize
+ * the robot. The UltrasonicLocalizer class is first implemented to orient the robot at the required
+ * heading. Then, upon the user's prompt, the LightLocalizer class is implemented to drive the robot
+ * to the origin of the grid system.
+ * 
+ * @author1 Cristian Ciungu
+ * @author2 Hao Shu
+ * @version 12-02-2019
+ * 
+ */
+package ca.mcgill.ecse211.lab4;
+
+import lejos.hardware.Button;
+import lejos.hardware.ev3.LocalEV3;
+import lejos.hardware.lcd.TextLCD;
+import lejos.hardware.motor.EV3LargeRegulatedMotor;
+import lejos.hardware.port.Port;
+import lejos.hardware.sensor.EV3UltrasonicSensor;
+import lejos.hardware.sensor.SensorModes;
+import lejos.robotics.SampleProvider;
+import ca.mcgill.ecse211.odometer.Odometer;
+import ca.mcgill.ecse211.odometer.OdometerExceptions;
+
+
+
+public class Lab4 {
+
+  // Motor Objects, and Robot related parameters
+  private static final EV3LargeRegulatedMotor leftMotor =
+      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("A"));
+  private static final EV3LargeRegulatedMotor rightMotor =
+      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("B"));
+  private static final TextLCD lcd = LocalEV3.get().getTextLCD();
+  private static final Port usPort = LocalEV3.get().getPort("S1");
+
+  // Robot related parameters
+  public static final double WHEEL_RAD = 2.2;// the radius of the wheel
+  public static final double TRACK = 10.6;// the axel track of the robot

```

Comment:
ALL of the above (including the motors) must be individually Javadoc'ed, like this:

```java
/**
 * The left motor.
 */
public static final EV3LargeRegulatedMotor LEFT_MOTOR =
      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("A"));

/**
 * The right motor.
 */
public static final EV3LargeRegulatedMotor RIGHT_MOTOR =
      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("B"));

/**
 * The LCD screen.
 */
public static final TextLCD LCD = LocalEV3.get().getTextLCD();

// ...

/**
 * The axle track of the robot.
 */
public static final double TRACK = 10.6;
```

Also note that the hardware resources should be **`public`** and they should be in `ALL_CAPS` because they are CONSTANTS.